onnx/backend/test/case/node/__init__.py

```diff
@@ -155,6 +156,14 @@ def expect(node,  # type: onnx.NodeProto
         inputs=inputs_vi,
         outputs=outputs_vi)
     kwargs[str('producer_name')] = 'backend-test'
+
+    if 'opset_imports' not in kwargs:
+        # To make sure the model will be produced with the same opset_version after opset changes
+        # By default, it uses since_version as opset_version for produced models
+        since_version = onnx.defs.get_schema(node.op_type, node.domain).since_version
+        produce_opset_version = int(_TargetOpsetVersion) if _TargetOpsetVersion is not None else since_version  # type: ignore
```

Comment:
Instead of a random target opset version the option should be to use the max inclusive version for the test... If we use a random opset version most likely we won't be able to generate test data for all tests this is because if users choose an older opset version for which the tests are not applicable we have to skip those tests.

Reply:
Also add comments on line 129 regarding this change